upb/def.c

```diff
@@ -364,19 +387,24 @@ bool upb_def_freeze(upb_def *const* defs, int n, upb_status *s) {
    * limit. */
   maxdepth = UPB_MAX_MESSAGE_DEPTH * 2;
 
-  /* Validation all passed; freeze the defs. */
-  ret = upb_refcounted_freeze((upb_refcounted * const *)defs, n, s, maxdepth);
```

Comment:
Now that I read the rest I see that the first add_n are all upb_defs but that the last freeze_n - add_n can be any refcounted object.
I think an intenral typed method taking defs to prepare for freezing would be cleaner:

 upb_freeze would be:
upb_def_prepare_for_freeze(defs, n);
upb_refcounted_freeze(defs, n);

and symtab_add would be:
upb_def_prepare_for_freeze(defs, n)
upb_refcounted_freeze(refcount_objs /*defs and file*/, freeze_n);